var/spack/repos/builtin/packages/fms/package.py

```diff
@@ -13,25 +13,45 @@ class Fms(CMakePackage):
     system models."""
 
     homepage = "https://github.com/NOAA-GFDL/FMS"
-    url      = "https://github.com/NOAA-GFDL/FMS/archive/refs/tags/2021.02.01.tar.gz"
-
-    maintainers = ['kgerheiser', 'Hang-Lei-NOAA', 'edwardhartnett']
-
-    variant('64bit', default=True, description='64 bit?')
-    variant('gfs_phys', default=True, description='Use GFS Physics?')
-    variant('openmp', default=True, description='Use OpenMP?')
-
-    version('2021.04', sha256='dcb4fe80cb3b7846f7cf89b812afff09a78a10261ea048a851f28935d6b241b1')
+    url      = "https://github.com/NOAA-GFDL/FMS/archive/refs/tags/2022.02.tar.gz"
+    git      = "https://github.com/NOAA-GFDL/FMS.git"
+
+    maintainers = ['kgerheiser', 'Hang-Lei-NOAA', 'edwardhartnett', 'rem1776']
+
+    variant('64bit', default=True, description='Build a version of the library with default 64 bit reals')
+    variant('gfs_phys', default=True, description='Use GFS Physics')
+    variant('openmp', default=True, description='Use OpenMP')
+    variant('quad_precision', default=True, description='quad precision reals')
+    variant('yaml', default=False, description='yaml input file support(requires libyaml)', when='@2021.04')
+    variant('constants', default='GFDL', description='Build with <X> constants parameter definitions',
+            values=('GFDL','GEOS','GFS'), multi=False, when='@2022.02')
+    variant('fpic', default=False, description='Build with position independent code', when='@2022.02')
+    version('2022.02',    sha256='ad4978302b219e11b883b2f52519e1ee455137ad947474abb316c8654f72c874')
```

Comment:
It's common practice in all other packages to have versions before variants. Can we do the same here?